**Install** Library yang dibutuhkan

In [ ]:
!pip install apyori

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
  Created wheel for apyori: filename=apyori-1.1.2-py3-none-any.whl size=5973 sha256=aff728cccae747056217e3eaa6e7362b12498157a04e3e02177c2edf7c182670
  Stored in directory: /root/.cache/pip/wheels/1b/02/6c/a45230be8603bd95c0a51cd2b289aefdd860c1a100eab73661
Successfully built apyori


Hubungkan ke Google Drive

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


Import library


In [ ]:
# Memanggil library yang dibutuhkan
import pandas as pd
import numpy as np
from apyori import apriori

Proses Perhitungan Apriori

Persiapan *Dataset*



In [ ]:
# Loading the Data
df= pd.read_excel('/content/drive/MyDrive/dataset_penjualan_2022_2023.xls')
df.head()


Tanggal ID Transaksi  \
0 2022-09-02          A01   
1 2022-09-03          A02   
2 2022-09-04          A03   
3 2022-09-05          A04   
4 2022-09-06          A05   

                                                                                                                                                                                                                                                                                                                                                                 Items  
0                                                                                                                                                                                                                                                                                                                                               Pepsodent, Gulaku 50gr  
1                                                                                                                                                                 Molto , Daya,Surya JR Rokok 50gr,Samsu Premium JR Rokok 100gr,Sirup ABC,Aice rasa Jagung,Indomie Goreng,Violin 40gr,GAYO,Guntur Simadu,Bebelac,Sari Gandum,belVita,Regal,WRP,Buffalo Polos,Teh Pucuk  
2                                                                                                    Masker 4D Premium,Pengesat Natural,Wet Tissue,Roll Tissue,Coconut Water,Garnier Men,Wardah Moisturizer,Kao Biore,SASA,Indomie Goreng,Walss,Rexona,Sasha,Clear Sampho,Kobe,Pop Mie,Kimbo,Lemonilo,Mars Brand,Jamaica Esse Duren Simadu Menthol,Amplang,Ppir Naraya  
3  Sandal,Hand Wash,Pepsodent,Sedaap Mie,Mascara Maybelline,Ultramilk Coklat,Marlboro Merah JR Rokok 100gr,Jago Mole 333 Hijau,melon Jago rasa New50,Indomie Goreng,Violin 40gr,Whiskey Duren Simadu,Green tea Duren Simadu,Jamaica Esse Duren Simadu,Nuget So Good,Bebelac,Sari Gandum,WRP,Teh Pucuk,Mars Brand,Jamaica Esse Duren Simadu Menthol,Amplang,Ppir Naraya  
4                                                                                                                                                                                                               Masker 4 pcs,Adem Sari,Sampoerna Mild JR Rokok 50gr,Samsu Premium JR Rokok 50gr,Jago Mole 333 Hijau,Strawberry Millky SW PREMIUM,Sari Gandum,Teh Pucuk

Cleaning Data

In [ ]:
# Membuang kolom Tanggal, ID Transaksi
data=df.drop(['Tanggal','ID Transaksi'],axis=1)


Membuat Data Training

In [ ]:
# Membuat list dalam list dari transaksi pembelian barang
records = []
for i in range(data.shape[0]):
    records.append([str(data.values[i,j]).split(',') for j in range(data.shape[1])])

trx = [[] for trx in range(len(records))]
for i in range(len(records)):
    for j in records[i][0]:
        trx[i].append(j)
trx

[['Pepsodent', ' Gulaku 50gr'],
 ['Molto ',
  ' Daya',
  'Surya JR Rokok 50gr',
  'Samsu Premium JR Rokok 100gr',
  'Sirup ABC',
  'Aice rasa Jagung',
  'Indomie Goreng',
  'Violin 40gr',
  'GAYO',
  'Guntur Simadu',
  'Bebelac',
  'Sari Gandum',
  'belVita',
  'Regal',
  'WRP',
  'Buffalo Polos',
  'Teh Pucuk'],
 ['Masker 4D Premium',
  'Pengesat Natural',
  'Wet Tissue',
  'Roll Tissue',
  'Coconut Water',
  'Garnier Men',
  'Wardah Moisturizer',
  'Kao Biore',
  'SASA',
  'Indomie Goreng',
  'Walss',
  'Rexona',
  'Sasha',
  'Clear Sampho',
  'Kobe',
  'Pop Mie',
  'Kimbo',
  'Lemonilo',
  'Mars Brand',
  'Jamaica Esse Duren Simadu Menthol',
  'Amplang',
  'Ppir Naraya'],
 ['Sandal',
  'Hand Wash',
  'Pepsodent',
  'Sedaap Mie',
  'Mascara Maybelline',
  'Ultramilk Coklat',
  'Marlboro Merah JR Rokok 100gr',
  'Jago Mole 333 Hijau',
  'melon Jago rasa New50',
  'Indomie Goreng',
  'Violin 40gr',
  'Whiskey Duren Simadu',
  'Green tea Duren Simadu',
  'Jamaica Esse Duren Simadu',
  '

Lakukan Perhitungan *Apriori*

In [ ]:
# Menggunakan fungsi apriori untuk membuat asosiasi
association_rules = apriori(trx, min_support=0.14, min_confidence=0.80,min_lift=1)
# Membuat list hasil dari algoritma apriori
association_results = association_rules

**Menampilkan** Hasil Perhitungan

In [ ]:
# Menampilkan hasil asosiasi dari item 
pd.set_option('max_colwidth', 1000)
Result=pd.DataFrame(columns=['Rule','Support','Confidence'])
for item in association_results:
    pair = item[2]
    for i in pair:
        items = str([x for x in i[0]])
        if i[3]!=1:
            Result=Result.append({
                'Rule':str([x for x in i[0]])+ " -> " +str([x for x in i[1]]),
                'Support':str(round(item[1]*100,2))+'%',
                'Confidence':str(round(i[2] *100,2))+'%'
                },ignore_index=True)
Result


Rule Support Confidence
0  ['Tango'] -> ['Teh Pucuk']  21.21%      87.5%

kesimpulan dari penerapan Association Rule menggunakan Algoritma Apriori dari hasil percobaan dari menggunakan pemrograman python dari 100 data transaksi penjualan adalah sebagai berikut ini :

Jika membeli Tango maka akan juga membeli Teh Pucuk dengan memiliki confidence atau tingkat kepercayaan sebesar 87.5%.